Expose all types directly from the root - fixes #71

bartcallant opened the pull request:
I've exported every thing through a barrel file in the specific folder. I opted to not use `export * from  './file'` to avoid naming collisions.

But by doing this I've actually found a naming collision in the types, the top `ParseResult` is exported in both `src/lib/utils/query.ts` and `src/lib/utils/update.ts`.

Since the types are not equal I am not sure what to do.

bartcallant commented:
@devinmotion I could indeed do that, but personally I'm not the biggest fan of that. Because you lose the sense of what is actually exported. I am willing to change it, let's see what @SamVerschueren think about this.

bartcallant commented:
@SamVerschueren could you also think about @devinmotion's [comment](https://github.com/SamVerschueren/dynongo/issues/71#issuecomment-538779158)?

MrCube42 commented:
@bartcallant Wasn't sure whether you knew about this possibility. Obviously you are ;) and you are right: You would give up some control from the top down to the bottom of the code base.

Thanks for your explaination. Decision is up to @SamVerschueren.

Have a nice #Hacktoberfest :)

SamVerschueren commented on src/index.ts:
```diff
@@ -1,3 +1,5 @@
+import { DynamoDB } from './lib';
 
-export = new DynamoDB();
+export { ReadItem, TransactRead, TransactQuery, TransactDeleteItem, TransactInsertItem, TransactUpdateItem, WriteItem, TransactWrite, TransactMethod, BaseQuery, CreateTable, DeleteItem, DeleteTable, Executable, InsertItem, ListTables, Method, Query, Scan, UpdateItem, Map, Params, QueryBuilder, ThroughPut, AttributeDefinition, KeySchema, Projection, GlobalSecondaryIndex, LocalSecondaryIndex, Schem ...
```
Can you spread this over multiple lines? I believe it improves readability.

Not sure about exporting everything though. I think the method's on their own would be a good start.

If it's not directly usable publicly, we shouldn't export it. I believe the `ParseResult` and the `ValueNameResult` are types which are only used internally if I'm not mistaken?

SamVerschueren commented on src/test/test.ts:
```diff
@@ -1,5 +1,5 @@
 import test from 'ava';
-import db = require('../');
+import db from '../';
```
```suggestion
import db from '..';
```

bartcallant commented:
I've addressed the comments. 

These are the currently exported types:
```
ReadItem,
TransactRead,
TransactQuery,
TransactDeleteItem,
TransactInsertItem,
TransactUpdateItem,
WriteItem,
TransactWrite,
TransactMethod,
BaseQuery,
CreateTable,
DeleteItem,
DeleteTable,
Executable,
InsertItem,
ListTables,
Method,
Query,
Scan,
UpdateItem,
Map,
UpdateQuery,
DynamoDBOptions,
DynamoDB,
TableOptions,
Table
```

Let me know if there are more which can be excluded from this list.

SamVerschueren commented on src/index.ts:
```diff
@@ -1,3 +1,32 @@
+	Query,
+	Scan,
+	UpdateItem,
+	Map,
```
Where is this one used? I don't really like how I named this one as it's a colission with the JavaScript `Map`. So if we could keep it internal, I would love to do that because I think I want to rename this one.

bartcallant replied:
It is used for `ExpressionAttributeNames`, `ExpressionAttributeValues` and `Table.insert()`

SamVerschueren commented:
Thanks @bartcallant!
